test/integration/033_event_tracking_test/test_events.py

```diff
@@ -0,0 +1,631 @@
+
+
+class TestEventTracking(DBTIntegrationTest):
+    maxDiff = None
+
+    @property
+    def profile_config(self):
+        return {
+            'config': {
+                'send_anonymous_usage_stats': True
+            }
+        }
+
+    @property
+    def schema(self):
+        return "event_tracking_033"
+
+    @staticmethod
+    def dir(path):
+        return "test/integration/033_event_tracking_test/" + path.lstrip("/")
+
+    @property
+    def models(self):
+        return self.dir("models")
+
+    # TODO : Handle the subject. Should be the same every time!
+    # TODO : Regex match a uuid for user_id, invocation_id?
+
+    @mock.patch('dbt.tracking.tracker.track_struct_event')
+    def run_event_test(
+        self,
+        cmd,
+        expected_calls,
+        expected_contexts,
+        track_fn,
+        expect_pass=True,
+        expect_raise=False
+    ):
+        track_fn.reset_mock()
+
+        project_id = hashlib.md5(
+            self.project['name'].encode('utf-8')).hexdigest()
+        version = str(dbt.version.get_installed_version())
+
+        if expect_raise:
+            with self.assertRaises(BaseException):
+                self.run_dbt(cmd, expect_pass=expect_pass)
+        else:
+            self.run_dbt(cmd, expect_pass=expect_pass)
+
+        user_id = dbt.tracking.active_user.id
+        invocation_id = dbt.tracking.active_user.invocation_id
+
+        self.assertTrue(len(user_id) > 0)
+        self.assertTrue(len(invocation_id) > 0)
+
+        track_fn.assert_has_calls(expected_calls)
+
+        ordered_contexts = []
+
+        for (args, kwargs) in track_fn.call_args_list:
+            ordered_contexts.append(
+                [context.__dict__ for context in kwargs['context']]
+            )
+
+        populated_contexts = []
+
+        for context in expected_contexts:
+            if callable(context):
+                populated_contexts.append(context(
+                    project_id, user_id, invocation_id, version))
+            else:
+                populated_contexts.append(context)
+
+        self.assertEqual(
+            ordered_contexts,
+            populated_contexts
+        )
+
+    def build_context(
+        self,
+        command,
+        progress,
+        result_type=None
+    ):
+
+        def populate(
+            project_id,
+            user_id,
+            invocation_id,
+            version
+        ):
+            return [
+                {
+                    'schema': 'iglu:com.dbt/invocation/jsonschema/1-0-0',
+                    'data': {
+                        'project_id': project_id,
+                        'user_id': user_id,
+                        'invocation_id': invocation_id,
+                        'version': version,
+
+                        'command': command,
+                        'progress': progress,
+                        'run_type': 'regular',
+
+                        'options': None,  # TODO : Add options to compile cmd!
```

Comment:
@cmcarthur I think we should try running a cmd like:
```
dbt compile --vars 'abc: 123'
```

and then confirm that the CLI opts are not tracked. This is the currently implemented behavior, but I want to make sure we don't actually track these things in the future